app/scripts/metamask-controller.js

```diff
@@ -811,12 +811,21 @@ export default class MetamaskController extends EventEmitter {
         const hasActiveShieldSubscription = getIsShieldSubscriptionActive(
           state.subscriptions,
         );
-
         if (hasActiveShieldSubscription) {
```

Comment:
```suggestion
        const { useExternalServices } = this.preferencesController.state;
        if (hasActiveShieldSubscription && useExternalServices) {
```

We should check before we fetching claims.